components/automate-ui/src/app/components/page-picker/page-picker.component.ts

```diff
@@ -54,6 +54,7 @@ export class PagePickerComponent implements OnChanges {
 
   ngOnChanges() {
     this.last = Math.ceil(this.total / this.perPage) || 1;
+    this.page = this.page > this.last ? this.last : this.page;
```

Comment:
This was the main culprit, the current page data wasn't being updated, which was throwing off the math below creating selectable pages that didn't actually exist.  This makes sure that if we're dropping to a fewer amount of pages, the current page is adjusted to reflect that change.

Reply:
Good, but can simplify a bit:
```suggestion
    this.page = Math.min(this.page, this.last);
```